torchcde/interpolation_hermite_cubic_bdiff.py

```diff
@@ -0,0 +1,60 @@
+import torch
+from torchcde.interpolation_cubic import NaturalCubicSpline
+from torchcde.interpolation_linear import linear_interpolation_coeffs
+
+
+def _setup_hermite_cubic_coeffs_w_backward_differences(times, coeffs, derivs, device=None):
+    """Compute backward hermite from linear coeffs."""
+    x_prev = coeffs[..., :-1, :]
+    x_next = coeffs[..., 1:, :]
+    # Let x_0 - x_{-1} = x_1 - x_0
+    derivs_prev = torch.cat((derivs[..., [0], :], derivs[..., :-1, :]), axis=-2)
+    derivs_next = derivs
+    x_diff = x_next - x_prev
+    t_diff = (times[1:] - times[:-1]).unsqueeze(-1)
+    # Coeffs
+    a = x_prev
+    b = derivs_prev
+    c = (1 / t_diff ** 2) * (3 * (x_diff - b * t_diff) - t_diff * (derivs_next - derivs_prev))
```

Comment:
It would be preferred not to multiply then divide by `t_diff`. (For numerical stability.)